### PR TITLE
Create the Search Layout: SearchResultList and SearchResultListItem

### LINKED ISSUE #37: Create Destination Pages
# User story
As a user, 
I want to see the main page of the destination of my choice
so that I find exactly what I want.

## Acceptance criteria 

- [x] Create the destination page template
- [x] Decide the metadata required in every destination (city and category)

### LINKED ISSUE #43: Apply Logic to Article / Destination
# User story
As a **developer**, 
I want to create the map function to take the data from my articles collection
so that the articles are generated every time I click on their image.

## Notes / Links

https://es.reactjs.org/docs/dom-elements.html

## Acceptance criteria 

- [x] Link the articles of the first carousel in Menu.js
- [x] Link the destination pages of the second carousel
- [x] Link the articles of the thirst carousel in Menu.js
- [x] Create the filter in the Search.js

### PR DESCRIPTION
**Pull request by manuelsanchez2**
Hi, in this pull request I focused on the following steps:

- [X] Layout of the Search Page - reviewed by Slawo (thanks!)
- [X] Gathering of some data regarding Articles
- [X] Creation of Articles.js to export a function to call the api and fetch the articles
- [X] Use the map method to display all the results in the Search.js

Following steps are:

- [x] Apply the filter logic using cities and categories
- [x] Link with the page of every article

Screenshot 2020-09-23 at 17 35 10

### CODE REVIEW

File: client/src/components/SearchResultList/SearchResultList.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import styled from "@emotion/styled";
+
+const SearchResultList = styled.nav``;
+
+export default SearchResultList;
```
Comment (slawoe) [inline client/src/components/SearchResultList/SearchResultList.js:5]: Why are you exporting an unstyled nav as styled Component?
Isn't it simpler just to use a nav in your list?

File: client/src/components/SearchResultListItem/SearchResultListItem.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      <Link to="/destinations/hamburg">
+        <img src="https://i.imgur.com/Jycf1xG.jpg" alt="" />
+        <div>
+          <h4>Alojamiento en Hamburgo</h4>
+          <small>
+            El tema de cómo llegar en avión o cómo moverse por la ciudad o el
+            país es importante, está claro, pero queremos tener la certeza de
+            que descansaremos sobre techo firme y de que nuestras maletas
+            estarán a salvo en algún lugar.
+          </small>
```
Comment (slawoe) [inline client/src/components/SearchResultListItem/SearchResultListItem.js]: Maybe use a paragraph and size it with em.
Or is it sure you want to use small because of semantic reasons?

**Review by slawoe — COMMENTED** [review]
LGTM, but I have little questions about styled component logic and semantic.
Keep the good work!

File: client/src/components/SearchResultListItem/SearchResultListItem.js (hunk 2 above)
Comment (manuelsanchez2) [reply client/src/components/SearchResultListItem/SearchResultListItem.js]: Yeah, it could be also paragraph :)

File: client/src/components/SearchResultList/SearchResultList.js (hunk 1 above)
Comment (manuelsanchez2) [reply client/src/components/SearchResultList/SearchResultList.js:5]: I love complicating my life, and well... the name is cool! thanks!

File: client/src/components/SearchResultList/SearchResultList.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import styled from "@emotion/styled";
+
+const SearchResultList = styled.nav``;
```
Comment (karstenlenz) [inline client/src/components/SearchResultList/SearchResultList.js:3]: Do you really need this styled component if you don't add any styles to it? Can't you just use a normal HTML-nav element?

File: client/src/components/SearchResultListItemImage/SearchResultListItemImage.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const { default: styled } = require("@emotion/styled");
```
Comment (karstenlenz) [inline client/src/components/SearchResultListItemImage/SearchResultListItemImage.js]: Why don't you use "import" instead of require like in the other files?

File: client/src/components/SearchResultListItemImage/index.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export { default } from "./SearchResultListItemImage";
```
Comment (karstenlenz) [inline client/src/components/SearchResultListItemImage/index.js:1]: I noticed that you have a lot of index.js files that only export something. Are these files really necessary? Can't you just do the import from "./SearchResultListItemImage" directly?

**Review by karstenlenz — APPROVED** [review]
Senor Sanchez, su code es muy bueno! I couldn't find much to comment. Just left some questions on parts I wasn't sure about. I hope the comments are helpful.

File: client/src/components/SearchResultList/SearchResultList.js (hunk 3 above)
Comment (manuelsanchez2) [reply client/src/components/SearchResultList/SearchResultList.js:3]: Yes, you are right! I did it in case later I added some styling, but I could simple use a nav :) thanks!

File: client/src/components/SearchResultListItemImage/SearchResultListItemImage.js (hunk 4 above)
Comment (manuelsanchez2) [reply client/src/components/SearchResultListItemImage/SearchResultListItemImage.js]: Wow! Thanks for pointing this out! It is weird, I thought I dit the auto-import and something went wrong! I will fix it!

File: client/src/components/SearchResultListItemImage/index.js (hunk 5 above)
Comment (manuelsanchez2) [reply client/src/components/SearchResultListItemImage/index.js:1]: To be honest, I do not really know the reason why, I just saw how Leon did it in his component during the speed run with the logs project, and that's why I do It like that. But I will ask him tomorrow and tell you if there is a good reason to do It!

File: client/src/components/DestinationArticleImage/DestinationArticleImage.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        <img src={src} alt={alt} />
+        <small>{imageTitle}</small>
+      </Link>
+    </Container>
+  );
+};
+
+DestinationArticleImage.propTypes = {
+  children: PropTypes.node,
+  to: PropTypes.any,
```
Comment (aruppert) [inline client/src/components/DestinationArticleImage/DestinationArticleImage.js]: I wonder if this is necessary? My es-lint is fine without it...  :)

**Review by aruppert — APPROVED** [review]
lgtm :)

File: client/src/components/DestinationArticleImage/DestinationArticleImage.js (hunk 6 above)
Comment (manuelsanchez2) [reply client/src/components/DestinationArticleImage/DestinationArticleImage.js]: You are right! Thank you so much :)
